pages/applications/[applicationId].tsx

```diff
@@ -71,14 +76,37 @@ function Application() {
     }
   }, [shouldFetchData]);
 
+  const onClickCreateNewTask = () => setIsAddingNewTask(true);
+
+  const onClickTask = (taskData: TaskData) => {
+    setSelectedTask(taskData);
+  };
+
   return (
     <Spinner isLoading={isLoading}>
       {hasSuccessfullyFetchedApplication && timelineItems.length === 0 && (
         <div className="mr-5 ml-5 flex justify-center text-center text-gray-300">
-          This application seems very empty. Add you first stage or task now!
+          This application seems very empty. Add your first stage or task now!
         </div>
       )}
 
+      {isAddingNewTask && applicationId && (
+        <NewTaskModal
+          applicationId={applicationId}
+          setIsAddingNewTask={setIsAddingNewTask}
+          setShouldFetchData={setShouldFetchData}
+        />
+      )}
+
+      {selectedTask && applicationId && (
+        <EditTaskModal
+          applicationId={applicationId}
+          initialTask={selectedTask}
+          setSelectedTask={setSelectedTask}
+          setShouldFetchData={setShouldFetchData}
+        />
+      )}
```

Comment:
Is it better for the `XModal` to take in a `isOpen` prop and we use a useState here for the state management.

We can also return null if `applicationId` is not defined.

Suggesting based on HY's approach in `pages/applications/create.tsx` in https://github.com/sivayogasubramanian/caza/pull/55

Reply:
Rather not because once that happens I can pass in nullable task and application id and that makes the checks in `EditTaskModal` more messy.